api/repository/transaction/index.ts

```diff
@@ -64,17 +63,10 @@ class TransactionPGRepository {
             offset,
         });
 
-
         if (transactions && transactions.length) {
-            // TODO: optimize and refactor this
-            let transactionsCount: number;
-            if (isFiltered(filter)) {
-                isFiltered(filter, new Set(['type']))
-                    ? transactionsCount = DEFAULT_COUNT
-                    : transactionsCount = Number(transactions[0].count);
-            } else {
-                transactionsCount = this.transactionsCount;
-            }
+            const transactionsCount: number = transactions[0].count
+                ? Number(transactions[0].count)
```

Comment:
Broken logic
Old logic include fake count for transaction filtered not only by type